src/client/httpClient.ts

```diff
@@ -249,18 +244,39 @@ export const handleIfErrorResponse = (response: any, data: Record<string, any>)
 
 function serializeCommand(data: Record<string, any>, pretty?: boolean): string {
   if (pretty) {
-    return JSON.stringify(data, function(key, value) {
-      if (typeof value === 'bigint') {
+    return EJSON.stringify(data, function (key, value) {
+      if (value != null && typeof value === 'bigint') {
         return Number(value);
+      } else if (value != null && typeof value === 'object') {
+        // ObjectId to strings
+        if (value.$oid) return value.$oid;
+        else if (value.$date) {
+          // Use numbers instead of strings for dates
+          value.$date = new Date(value.$date).valueOf();
+        }
       }
       return value;
     }, '  ');
   }
-  return JSON.stringify(data, function(key, value) {
-    if (typeof value === 'bigint') {
+  return EJSON.stringify(data, function (key, value) {
+    if (value != null && typeof value === 'bigint') {
       return Number(value);
+    } else if (value != null && typeof value === 'object') {
+      // ObjectId to strings
+      if (value.$oid) return value.$oid;
+      else if (value.$date) {
+        // Use numbers instead of strings for dates
+        value.$date = new Date(value.$date).valueOf();
+      }
     }
     return value;
   });
 }
 
+function deserialize(data: Record<string, any>): Record<string, any> {
+  if (data != null && data != undefined) {
```

Comment:
Minor nitpick `if (date != null)` catches both null and undefined, so the `&& data != undefined` is redundant.

Reply:
Fixed in this commit [44e5864](https://github.com/stargate/stargate-mongoose/pull/91/commits/44e5864318d3afdad173dd0e2eafe69f0de4cb4f)